### PR TITLE
implement #23

### PR DESCRIPTION
**Pull request by ershovd**
- implement messaging from FlagTaskDialogViewModel to MainPageViewModel
- implement running parametrised command UnwindFlaggedTaskCommand with Task
- animate to default viewcell position in SwipableViewCellRenderer

Things to think about:
1. Subscribing for listview event is done in every cell `FindParent().UnWindCell += `  may be not optimal approach, no unsubscribing 
2. the solution require iOS specific implementation, looking for options to make 100% portable (if possible).

### CODE REVIEW

**Comment by j0nscalet** [comment]
@ershovd I just ran this branch on my side and the functionality is perfect. Really good job with this Dmitry. 👍 

I have a quick question on a tweak to this and wanted to see what you think. [Watch this video](https://drive.google.com/open?id=0B8X_VSa95wMwSTVzYkpFb0RWU3M)

File: QualityKitchen.iOS/Renderers/SwipableViewCellRenderer.cs [hunk 1]
```diff
@@ -107,6 +118,14 @@ private void HandleDraggingLeft (UITableViewCell cell, double distance)
 			_rightLabel.Frame = new CGRect (cell.Frame.Width, cell.Bounds.Y, -distance, cell.Frame.Height);
 		}
 
+		private void AnimateToInitialPosition(UITableViewCell cell)
```
Comment (j0nscalet) [inline QualityKitchen.iOS/Renderers/SwipableViewCellRenderer.cs]: Great job refactoring this animation into a method. Though it's small I appreciate your attentiveness to write clean code. 😃

File: QualityKitchen.Mobile/Application/Views/SwipableListView/SwipableListView.cs [hunk 2]
```diff
@@ -21,14 +31,30 @@ public void OnLeftSwipe (SwipeEventArgs e)
...
 		}
 
-		public class SwipeEventArgs : EventArgs
+		public void UndoSwipe(Domain.Model.Task task)
 		{
-			public SwipeEventArgs (SwipableViewCell viewCell)
+			var tasks = ItemsSource.Cast<Domain.Model.Task>().ToList();
+
+			// find row index of SwipableViewCellViewCell 
+
```
Comment (j0nscalet) [inline QualityKitchen.Mobile/Application/Views/SwipableListView/SwipableListView.cs]: Did you mean to delete this line? Seems like you should.

File: QualityKitchen.Mobile/Application/ViewModels/MainPageViewModel.cs [hunk 3]
```diff
@@ -21,8 +21,17 @@ public MainPageViewModel ()
 		{
 			LoadBackgroundDataCommand = new Command(async () => await LoadDataInBackground());
 			TotalIncidents = 0;
+
+			MessagingCenter.Subscribe<FlagTaskDialogViewModel, TaskCancelledEventArgs>(this, FlagTaskDialogViewModel.cancelledTaskToken, UnwindFlaggedTask);
+		}
+
+		void UnwindFlaggedTask(FlagTaskDialogViewModel flagVM, TaskCancelledEventArgs taskEventArgs)
```
Comment (j0nscalet) [inline QualityKitchen.Mobile/Application/ViewModels/MainPageViewModel.cs]: - Use the full argument name here `dialogViewModel`
- Change `taskEventArgs` to `cancelTaskEventArgs`

File: QualityKitchen.Mobile/Application/ViewModels/FlagTaskDialogViewModel.cs [hunk 4]
```diff
@@ -8,7 +8,9 @@
 namespace QualityKitchen.Mobile.Application.ViewModels
 {
 	public class FlagTaskDialogViewModel : ViewModel
-	{	
+	{
+		internal static readonly string cancelledTaskToken = "TaskCancelled";
```
Comment (j0nscalet) [inline QualityKitchen.Mobile/Application/ViewModels/FlagTaskDialogViewModel.cs]: - `public` is fine here as there's no future plans to expose this assembly for consumption. 
- Capitalize cancelledTaskToken to `CancelledTaskToken` and update references

**Comment by ershovd** [comment]
@j0nscalet  thanks for the feedback. I will fix the minor stuff you've mention, and will try to get working the unwinding animation, you've mention in video

File: QualityKitchen.Mobile/Application/Views/SwipableListView/SwipableListView.cs (hunk 2 above)
Comment (ershovd) [inline QualityKitchen.Mobile/Application/Views/SwipableListView/SwipableListView.cs]: not clear about that, do you want me to inline variable or delete the comment? I've did both.

**Comment by ershovd** [comment]
@j0nscalet  I've apply edits according to your suggestions. The issue with animation seems to be more complex. I've tested, it appears that the animation is firing immediately (the callback for end of animation is fired). this may be due to the fact the animation and unwind event raised before fully loading the page. I need more time to solve this, if you feel that I should focus on that. Otherwise, I would continue to work on other tasks.

**Comment by j0nscalet** [comment]
@ershovd what I took a look that seemed to be the case as well. 

Let's pause on this for now and circle back on a future sprint. I think what we've got in place is really good.

**Comment by j0nscalet** [comment]
@ershovd created #57 we'll circle back to you. Again, great work.

**Pull request merged**